packages/opentelemetry-exporter-datadog/src/datadogSpanProcessor.ts

```diff
@@ -0,0 +1,203 @@
+      this.logger.debug('Max Queue Size reached, dropping span');
+      return;
+    }
+
+    const traceId = span.spanContext['traceId'];
+
+    if (!this._traces.has(traceId)) {
+      this._traces.set(traceId, []);
+      // this._traces.get(traceId).set(SPANS_KEY, [])
+      this._traces_spans_started.set(traceId, 0);
+      this._traces_spans_finished.set(traceId, 0);
+    }
+
+    const trace = this._traces.get(traceId);
+    const traceSpansStarted = this._traces_spans_started.get(traceId);
+
+    if (traceSpansStarted >= this._maxTraceSize) {
+      this.logger.debug('Max Trace Size reached, dropping span');
+      return;
+    }
+
+    trace.push(span);
+    this._traces_spans_started.set(traceId, traceSpansStarted + 1);
+  }
+
+  onEnd(span: ReadableSpan): void {
+    if (this._isShutdown) {
+      return;
+    }
+
+    const traceId = span.spanContext['traceId'];
+
+    if (!this._traces.has(traceId)) {
+      return;
+    }
+
+    const traceSpansFinished = this._traces_spans_finished.get(traceId);
+
+    this._traces_spans_finished.set(traceId, traceSpansFinished + 1);
+
+    if (this._isExportable(traceId)) {
+      this._check_traces_queue.add(traceId);
+      this._maybeStartTimer();
+    }
+  }
+
+  getTraceContext(span: ReadableSpan): string | undefined[] {
+    return [
+      id(span.spanContext['traceId']),
+      id(span.spanContext['spanId']),
+      span.parentSpanId ? id(span.parentSpanId) : undefined,
+    ];
+  }
+
+  /** Send the span data list to exporter */
+  private _flush() {
+    this._clearTimer();
+    if (this._check_traces_queue.size === 0) return;
+
+    this._check_traces_queue.forEach(traceId => {
+      // check again in case spans have been added
+      if (this._isExportable(traceId)) {
+        const spans = this._traces.get(traceId);
+        this._traces.delete(traceId);
+        this._traces_spans_started.delete(traceId);
+        this._traces_spans_finished.delete(traceId);
+        this._check_traces_queue.delete(traceId);
+        this._exporter.export(spans, () => {});
+      }
+    });
+  }
+
+  private _maybeStartTimer() {
+    if (this._timer !== undefined) return;
+
+    this._timer = setTimeout(() => {
+      this._flush();
+    }, this._bufferTimeout);
+    unrefTimer(this._timer);
+  }
+
+  private _clearTimer() {
+    if (this._timer !== undefined) {
+      clearTimeout(this._timer);
+      this._timer = undefined;
+    }
+  }
+
+  private _allSpansCount(traces: Map<string, number>): number {
+    // sum all started spans in all traces
+    return [...traces.values()].reduce((a, b) => a + b, 0);
+  }
+
+  private _isExportable(traceId: string) {
+    const trace = this._traces.get(traceId);
+
+    if (
+      !trace ||
+      !this._traces_spans_started.has(traceId) ||
+      !this._traces_spans_finished.has(traceId)
+    ) {
+      return;
+    }
+
+    return (
+      this._traces_spans_started.get(traceId) -
+        this._traces_spans_finished.get(traceId) <=
```

Comment:
Weird indenting here. Does the linter force this?

Reply:
yup, there's a few like this, the linter complains otherwise, this is the output of `npm run lint:fix` i believe.